core/CMakeLists.txt

```diff
@@ -208,32 +211,31 @@ target_link_libraries(userver-core-internal
 )
 
 
-add_library(userver-utest STATIC ${LIBUTEST_SOURCES})
-target_compile_definitions(userver-utest PUBLIC $<TARGET_PROPERTY:${PROJECT_NAME},COMPILE_DEFINITIONS>)
-
-target_link_libraries(userver-utest
-  PUBLIC
-    libgtest
-    libgmock
-	userver-core-internal
-    ${PROJECT_NAME}
-  PRIVATE
-    Boost::program_options
-)
-target_include_directories(userver-utest PUBLIC
-  ${CMAKE_CURRENT_SOURCE_DIR}/testing/include
-)
-target_include_directories(userver-utest PRIVATE
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/
-    ${CMAKE_CURRENT_SOURCE_DIR}/testing/src/
-)
-
-# Target with no need to use userver namespace, but includes require userver/
-add_library(yandex-userver-utest INTERFACE)
-target_link_libraries(yandex-userver-utest INTERFACE userver-utest)
+if (USERVER_IS_THE_ROOT_PROJECT)
+    add_library(userver-utest STATIC ${LIBUTEST_SOURCES})
```

Comment:
userver-utest is not a set of tests, but a library to simplify testing of code that uses userver

Example https://github.com/userver-framework/service_template/blob/802e2c3b5927521e2b145ed1b1ac7ef7b75ded35/CMakeLists.txt#L27

So the userver-utest should be build even if USERVER_IS_THE_ROOT_PROJECT is FALSE

Reply:
@apolukhin do you mean userver have to build with gtest and gbench in all cases?

Reply:
If the framework user plans to use unit tests - then yes, `userver-utest` would be handy.

However, if the user wish to have only functional tests, then it may be reasonable to skip building this target. I think we need USERVER_FEATURE_UTEST feature macro to disable the `userver-utest` build. PR would be welcomed :)